s4/clarity/configuration/udf.py

```diff
@@ -102,3 +102,16 @@ def _add_preset_internal(self, preset_value):
         preset_node = ETree.SubElement(self.xml_root, 'preset')
         preset_node.text = types.obj_to_clarity_string(preset_value)
 
+
+class Field(WrappedXml):
+    """
+    Represents a UDF Field definition in the Clarity configuration.
+
+    https://www.genologics.com/files/permanent/API/latest/data_cnf.html#field
```

Comment:
This isn't correct. A cnf:field is already defined in s4.clarity.configuration.udf, what you're defining here is a ptp:field or a protstepcnf:field. This class should probably not have the name Field, it should be something more unambiguous like the ProtocolStepField class you deleted.

Reply:
I just added then deleted a similar comment above. This is a reference to a field, not an actual field. Usually these types of references just resolve directly to the object. That's a little trickier here because the references are by name rather than by id. In addition there is additional information in some cases. I would be tempted to call this a FieldReference and have it implement a Field or Udf accessor that gets you to the Udf object.